core/storage/collection/gae_models.py

```diff
@@ -434,6 +433,63 @@ def export_data(cls, user_id):
         }
 
 
+class CollectionRightsAllUsersModel(base_models.BaseModel):
+    """Temporary storage model for all user ids ever mentioned in the collection
+    rights.
+
+    The id of each instance is the id of the corresponding collection.
+    """
+    # The user_ids of users who are (or were in history) members of owner_ids,
+    # editor_ids, voice_artist_ids or viewer_ids in corresponding rights model.
+    all_user_ids = ndb.StringProperty(indexed=True, repeated=True)
+
+    @staticmethod
+    def get_deletion_policy():
+        """CollectionRightsAllUsersModel are temporary model that will be
+        deleted after user migration.
+        """
+        return base_models.DELETION_POLICY.DELETE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether CollectionRightsAllUsersModel references the given
+        user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            cls.all_user_ids == user_id).get(keys_only=True) is not None
+
+    @staticmethod
+    def get_user_id_migration_policy():
+        """CollectionRightsAllUsersModel has multiple fields with user ID."""
+        return base_models.USER_ID_MIGRATION_POLICY.CUSTOM
+
+    @classmethod
+    def migrate_model(cls, unused_old_user_id, unused_new_user_id):
+        """This model is used to verify the user ID migration so it will be
+        filled by the AddAllUserIdsOneOffJob and AddAllUserIdsSnapshotsOneOffJob
+        after the migration is done, thus it shouldn't be migrated here.
```

Comment:
I had some trouble understanding this docstring. Can you explain what "here" means in "shouldn't be migrated here"? Also explain in more detail why the migration is unnecessary.

Also -- I am not sure I understand why we even need this and the above method if the model has deletion policy DELETE?

Reply:
Done. This method is needed because `USER_ID_MIGRATION_POLICY = CUSTOM` (and that is because we need the `verify_model_user_ids_exist` to be run.